src/treeffuser/_base_tabular_diffusion.py

```diff
@@ -432,7 +432,7 @@ def predict_distribution(
 
         y_samples = self._sample_without_validation(X=X, n_samples=n_samples, verbose=verbose)
 
-        n_samples, batch, _ = y_samples.shape
+        batch = y_samples.shape[2]
```

Comment:
@aagrande It looks like it should be `shape[1]`

Reply:
good catch